### PR TITLE
Apply yapf formatter.

### PR DESCRIPTION
**Pull request by sile**
This PR applies [yapf](https://github.com/google/yapf) formatter to Optuna source code.

Although this branch contains many changes produced by `yapf`,  the essential ones are only the followings:
- `setup.cfg`

### CODE REVIEW

**Comment by sile** [comment]
[FYI] `yapf(style=pep8)` is almost compatible with `autopep8` except the following diff:
```diff
--- original/./tests/integration_tests/test_lightgbm.py
+++ fixed/./tests/integration_tests/test_lightgbm.py
@@ -109,20 +109,20 @@
             'objective': 'binary',
             'metric': ['auc', 'binary_error']
         },
-               dtrain,
-               1,
-               verbose_eval=False,
-               nfold=2,
-               callbacks=[pruning_callback])
+            dtrain,
+            1,
+            verbose_eval=False,
+            nfold=2,
+            callbacks=[pruning_callback])
     else:
         lgb.train({
             'objective': 'binary',
             'metric': ['auc', 'binary_error']
         },
-                  dtrain,
-                  1,
-                  valid_sets=[dtest],
-                  valid_names=valid_names,
-                  verbose_eval=False,
-                  callbacks=[pruning_callback])
+            dtrain,
+            1,
+            valid_sets=[dtest],
+            valid_names=valid_names,
+            verbose_eval=False,
+            callbacks=[pruning_callback])
     return 1.0
--- original/./tests/integration_tests/test_xgboost.py
+++ fixed/./tests/integration_tests/test_xgboost.py
@@ -46,11 +46,11 @@
             'silent': 1,
             'objective': 'binary:logistic'
         },
-                  dtrain,
-                  1,
-                  evals=[(dtest, 'validation')],
-                  verbose_eval=False,
-                  callbacks=[pruning_callback])
+            dtrain,
+            1,
+            evals=[(dtest, 'validation')],
+            verbose_eval=False,
+            callbacks=[pruning_callback])
         return 1.0
 
     study = optuna.create_study(pruner=DeterministicPruner(True))
```

**Comment by g-votte** [comment]
As discussed offline in the Optuna dev team, we are not going to include yapf in the `checks` job. This is because there still be frequent format change with the library's update. Let's revisit the library after the format get enough stable.

File: .yapfignore [hunk 1]
```diff
@@ -0,0 +1 @@
+venv
```
Comment (g-votte) [inline .yapfignore]: We can delete this file.

File: setup.cfg [hunk 2]
```diff
@@ -7,3 +7,8 @@ max-line-length = 99
 max-line-length = 99
 statistics = True
 exclude = optuna/samplers/_hyperopt.py,venv,build
+
+# This section is for yapf.
+[yapf]
```
Comment (g-votte) [inline setup.cfg]: Please revert config changes for yapf.

File: .yapfignore (hunk 1 above)
Comment (sile) [reply .yapfignore]: Deleted: https://github.com/pfnet/optuna/pull/319/commits/5ebce4d2f7cc70d9e0949f1070d41f9db991d256

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -15,8 +15,8 @@ $ circleci build --job checks
 ```
 
 The above job contains following checkers:
-- [flake8](http://flake8.pycqa.org)
 - [autopep8](https://github.com/hhatto/autopep8)
+- [flake8](http://flake8.pycqa.org)
```
Comment (toshihikoyanase) [inline CONTRIBUTING.md]: I think the contents of this file are identical to those in master branch.
Please revert this file to master branch version.

**Pull request merged**